pkg/model/manifest.go

```diff
@@ -655,9 +655,56 @@ func Read(bytes []byte) (*Manifest, error) {
 }
 
 func (m *Manifest) validate() error {
+	err := m.sanitizeSvcNames()
+	if err != nil {
+		return err
+	}
+
 	return m.validateDivert()
 }
 
+func (m *Manifest) sanitizeSvcNames() error {
+	sanitizedServicesNames := make(map[string]string)
+	for devKey, dev := range m.Dev {
+		if shouldBeSanitized(devKey) {
+			sanitizedSvcName := sanitizeName(devKey)
+			if _, ok := m.Dev[sanitizedSvcName]; ok {
+				return fmt.Errorf("could not sanitize '%s'. Service with name '%s' already exists", devKey, sanitizedSvcName)
+			}
+			dev.Name = sanitizedSvcName
+			m.Dev[sanitizedSvcName] = dev
+			sanitizedServicesNames[devKey] = sanitizedSvcName
+			delete(m.Dev, devKey)
+		}
+	}
+
+	for buildKey, build := range m.Build {
+		if shouldBeSanitized(buildKey) {
+			sanitizedSvcName := sanitizeName(buildKey)
+			if _, ok := m.Build[sanitizedSvcName]; ok {
+				return fmt.Errorf("could not sanitize '%s'. Service with name '%s' already exists", buildKey, sanitizedSvcName)
+			}
+			m.Build[sanitizedSvcName] = build
+			sanitizedServicesNames[buildKey] = sanitizedSvcName
+			delete(m.Dev, buildKey)
+		}
+	}
```

Comment:
We should be already doing this as we sanitize the build services to inject the environment variables. My feeling is that we spreading this into different places. Not sure if it makes sense or not, but doing this just when reading the manifest would avoid to have to do it in the rest of places

Reply:
IIRC, at the moment the only place where we sanitize service names is when unmarshaling an stack. Any inconvenience to moving it to the manifest `serializer.go`?  @jLopezbarb

Reply:
No, we should probably do it on serialisation